tests/unit/s2n_stuffer_network_order_test.c

```diff
@@ -220,12 +220,12 @@ int main(int argc, char **argv)
 
         /* Should throw an error if reservation has wrong size */
         reservation.length = sizeof(uint64_t);
-        EXPECT_FAILURE_WITH_ERRNO(s2n_stuffer_write_reservation(reservation, 0), S2N_ERR_SIZE_MISMATCH);
+        EXPECT_FAILURE_WITH_ERRNO(s2n_stuffer_write_reservation(reservation, 0), S2N_ERR_PRECONDITION_VIOLATION);
```

Comment:
I was concerned about changing the error code here, but callers are using GUARD, and this isn't public API.

Reply:
Yeah I think the better approach is to have a `validate` function instead so the error codes are more obvious (see https://github.com/awslabs/s2n/pull/1990/files#diff-60fd20905d3fbd3c3484960ae88a212fR26)

Reply:
Yep, I'm tracking this request on a separate PR as this would require updates across all proofs. I can also open an GitHub issue to revert the error code once the other PR is merged.